kernel/hashlib.h

```diff
@@ -651,13 +693,14 @@ class dict
 		return !operator==(other);
 	}
 
-	unsigned int hash() const {
-		unsigned int h = mkhash_init;
-		for (auto &entry : entries) {
-			h ^= hash_ops<K>::hash(entry.udata.first);
-			h ^= hash_ops<T>::hash(entry.udata.second);
+	hash_state_t hash_acc(hash_state_t h) const {
+		hash_state_t st = h;
+		st = mkhash(entries.size(), st);
+		for (auto &it : entries) {
```

Comment:
I think is order sensitive now, so two `dict` which are equal will fail to have identical hashes.

Reply:
- [ ] put the order-agnostic xor back

I don't expect dicts of dicts or dict comparison to be common and therefore important for performance so that should be fine